### PR TITLE
Polishing up order listing screen with Facets

### PR DESCRIPTION
**Pull request by justinengle-krs**
Zeplin:
image
Implementation:
image

### CODE REVIEW

**Review by casperlotter — DISMISSED** [review]

File: src/components/Orders/ChannelBucket.svelte [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<script lang="ts">
+  import type { Bucket } from '@/lib/es/types';
+  import { useQuery } from '@sveltestack/svelte-query';
+  import { getContext, onMount } from 'svelte';
+
+  export let data: Bucket;
+  const dependencyContainer: any = getContext('order-col-dependencies');
+
+  onMount(() => {
+    if (!dependencyContainer) console.error('Ensure order-col-dependency context is set.');
```
Comment (matthampton-stock2shop) [inline src/components/Orders/ChannelBucket.svelte:10]: Is this debugging code?

File: src/components/Orders/ChannelCell.svelte [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  export let data: any;
+  const dependencyContainer: any = getContext('order-col-dependencies');
+
+  onMount(() => {
+    if (!dependencyContainer) console.error('Ensure order-col-dependency context is set.');
+  });
+
+  const channelData = useQuery(['channels-get'], () => dependencyContainer.channelsRepo.get());
+  $: channels = $channelData.data || [];
+  $: channelName = channels.find((c) => c.id.toString() == data.channel_id)?.description ?? '-';
```
Comment (matthampton-stock2shop) [inline src/components/Orders/ChannelCell.svelte:14]: Same comment here RE loading / isError

**Review by matthampton-stock2shop — CHANGES_REQUESTED** [review]
Not done with the review yet - but need to leave now - so will pick this up tomorrow...

@justinengle-krs Could you run chromatic on this one? Thanks!

**Comment by justinengle-krs** [comment]
> Not done with the review yet - but need to leave now - so will pick this up tomorrow...
> 
> @justinengle-krs Could you run chromatic on this one? Thanks!

https://www.chromatic.com/build?appId=63ef6cddab1dec384359e1cd&number=83

File: src/stories/components/70-Orders/20-OrdersGridRow.stories.svelte [hunk 3]
```diff
@@ -5,20 +5,29 @@
...
+  import { setContext } from 'svelte';
+  import MockChannelsRepo from '@/stories/test-data/channels/mock-channels-repo';
+  import { QueryClient, QueryClientProvider } from '@sveltestack/svelte-query';
 
+  const queryClient = new QueryClient();
   const orders = orderJson as SystemOrdersDataWrapper;
   const orderRow = orders.system_orders[0];
   const systemColumns = configLocal.system.elasticSearch.customerOrders.columns;
-  const columnsIdToShow = ['id', 'created', 'channel_order_code', 'status', 'total_display'];
+  const columnsIdToShow = ['id', 'created', 'channel_id', 'status', 'total_display'];
```
Comment (matthampton-stock2shop) [inline src/stories/components/70-Orders/20-OrdersGridRow.stories.svelte:16]: I know this is just a test really, but could we keep the channel_order_code and add in the channel_id next to it?

File: src/routes/Orders.svelte [hunk 4]
```diff
@@ -1,153 +1,5 @@
...
-              </tr>
-            {/each}
-          </tbody>
-        </Table>
-        <Pagination {index} />
-      {/if}
-    </div>
-  </div>
-</div>
+<OrdersList />
```
Comment (matthampton-stock2shop) [inline src/routes/Orders.svelte:5]: Could we add a storybook screen for this route with a bunch of orders in it?

**Review by matthampton-stock2shop — CHANGES_REQUESTED** [review]
@justinengle-krs Just a couple of things - and sorry for the back-and-forth on that error thing

File: src/components/Orders/ChannelBucket.svelte (hunk 1 above)
Comment (matthampton-stock2shop) [reply src/components/Orders/ChannelBucket.svelte:10]: @justinengle-krs What is this about? Do we need it? I feel like I'd rather have the thing raise an exception if it doesn't have the context that it needs - cos that would be a bug, right? 

Or just fail later when it gets a null reference ... (i.e. not even check - just believe!)

**Comment by justinengle-krs** [comment]
> @justinengle-krs Just a couple of things - and sorry for the back-and-forth on that error thing

https://www.chromatic.com/build?appId=63ef6cddab1dec384359e1cd&number=86

**Review by matthampton-stock2shop — APPROVED** [review]

**Pull request merged**